test-network-function/lifecycle/suite.go

```diff
@@ -424,38 +448,45 @@ func testGracePeriod(env *config.TestEnvironment) {
 	})
 }
 
-func testShutdown(env *config.TestEnvironment) {
+func test(env *config.TestEnvironment) {
 	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestShudtownIdentifier)
+	testfunction(env, testID, relativeShutdownTestDirectoryPath, relativeShutdownTestPath, "pre-stop")
+
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestLivenessIdentifier)
+	testfunction(env, testID, relativeLivenessTestDirectoryPath, relativeLivenessTestPath, "liveness")
+
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestReadinessIdentifier)
+	testfunction(env, testID, relativeReadinessTestDirectoryPath, relativeReadinessTestPath, "readiness")
+}
+func testfunction(env *config.TestEnvironment, testID, relativeNameTestDirectoryPath, relativeNameTestPath, name string) {
 	ginkgo.It(testID, ginkgo.Label(testID), func() {
 		failedPods := []*configsections.Pod{}
-		ginkgo.By("Testing PUTs are configured with pre-stop lifecycle")
+		ginkgo.By(fmt.Sprintf("Testing PUTs are configured with %s lifecycle", name))
 		for _, podUnderTest := range env.PodsUnderTest {
-			ginkgo.By(fmt.Sprintf("should have pre-stop configured %s/%s", podUnderTest.Namespace, podUnderTest.Name))
-			passed := shutdownTest(podUnderTest.Namespace, podUnderTest.Name, env.GetLocalShellContext())
+			ginkgo.By(fmt.Sprintf("should have %s configured %s/%s", name, podUnderTest.Namespace, podUnderTest.Name))
+			passed := functionTest(podUnderTest.Namespace, podUnderTest.Name, env.GetLocalShellContext(), relativeNameTestDirectoryPath, relativeNameTestPath, name)
 			if !passed {
```

Comment:
all the test functions should be defined and used inside var _ = Describe block

Reply:
why? function "test" defined inside var_(Describe block). and this functions call another function. @hamadise